packages/core/src/useForm.ts

```diff
@@ -97,8 +97,11 @@ export function useForm(opts?: FormOptions) {
       if (valueIdx === -1) {
         return;
       }
-
-      _values[fieldName].splice(valueIdx, 1);
+      if (Array.isArray(_values[fieldName])) {
+        _values[fieldName].splice(valueIdx, 1);
+      } else {
+        delete _values[fieldName];
+      }
```

Comment:
Thank you for finding and fixing the bug, but there is a small style issue. I prefer to use early returns instead of `else`. Can you refactor this bit here?

```suggestion
      if (Array.isArray(_values[fieldName])) {
        _values[fieldName].splice(valueIdx, 1);
        return;
      }
      delete _values[fieldName];
```